test/solidlsp/scala/test_scala_language_server.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import os
+
+import pytest
+
+from solidlsp.language_servers.scala_language_server import ScalaLanguageServer
+from solidlsp.ls_config import Language, LanguageServerConfig
+from solidlsp.ls_logger import LanguageServerLogger
+from solidlsp.settings import SolidLSPSettings
+
+
+@pytest.fixture(scope="module")
+def scala_ls():
+    # Define the path to your test Scala repository
+    repo_root = os.path.abspath("test/resources/repos/scala")
+
+    # Create a dummy logger
+    logger = LanguageServerLogger(logging.getLogger(__name__))
+
+    # Create a LanguageServerConfig for Scala
+    config = LanguageServerConfig(code_language=Language.SCALA)
+
+    # Create SolidLSPSettings
+    solidlsp_settings = SolidLSPSettings()
+
+    # Instantiate the ScalaLanguageServer
+    ls = ScalaLanguageServer(config, logger, repo_root, solidlsp_settings)
+
+    # Start the server and yield it for tests
+    with ls.start_server():
+        yield ls
+
+
+def test_scala_document_symbols(scala_ls):
```

Comment:
as mentioned above, pls add tests on finding references within the same file and across files

Reply:
https://github.com/oraios/serena/pull/528/commits/2577779cfcbe26a48a51bca6e2fafb652826633a

Reply:
see comment below